File: src/main/java/org/goafabric/personservice/controller/dto/Person.java
package org.goafabric.personservice.controller.dto;

import jakarta.annotation.Nullable;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import java.util.List;

public record Person (
    @Nullable String id,
    @Nullable Long version,
    @NotNull @Size(min = 3, max = 255) String firstName,
    @NotNull @Size(min = 3, max = 255) String lastName,
    //Temporal birthDate,
    List<Address> address) {
    /*
    public Person(String id,
                  Long version,
                  String firstName,
                  String lastName,
                  List<Address> address) {
        this(id, version, firstName, lastName, LocalDate.now(), address);
    }

     */
}

File: src/main/java/org/goafabric/personservice/controller/dto/Person.kt
package org.goafabric.personservice.controller.dto

import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Null
import jakarta.validation.constraints.Size


data class Person (
    val id: @Null String? = null,
    val firstName: @NotNull @Size(min = 3, max = 255) String,
    val lastName: @NotNull @Size(min = 3, max = 255) String,
    val address: Address?
)